eng/common/scripts/git-branch-push.ps1

```diff
@@ -41,33 +42,40 @@ param(
 # would fail the first time git wrote command output.
 $ErrorActionPreference = "Continue"
 
-if (!(git remote | ? {$_ -eq $RemoteName}))
+if ((git remote) -contains $RemoteName)
 {
-    Write-Host "git remote add $RemoteName $GitUrl"
-    git remote add $RemoteName $GitUrl
-    if ($LASTEXITCODE -ne 0)
-    {
-        Write-Error "Unable to add remote LASTEXITCODE=$($LASTEXITCODE), see command output above."
-        exit $LASTEXITCODE
-    }
+  Write-Host "git remote get-url $RemoteName"
+  $remoteUrl = git remote get-url $RemoteName
+  if ($remoteUrl -ne $GitUrl)
+  {
+    Write-Error "Remote with name $RemoteName already exists with an incompatible url [$remoteUrl] which should be [$GitUrl]."
+    exit 1
+  }
 }
-
-Write-Host "git fetch $RemoteName"
-git fetch $RemoteName
-if ($LASTEXITCODE -ne 0)
+else 
 {
-    Write-Error "Unable to fetch remote LASTEXITCODE=$($LASTEXITCODE), see command output above."
+  Write-Host "git remote add $RemoteName $GitUrl"
+  git remote add $RemoteName $GitUrl
+  if ($LASTEXITCODE -ne 0)
+  {
+    Write-Error "Unable to add remote LASTEXITCODE=$($LASTEXITCODE), see command output above."
     exit $LASTEXITCODE
+  }
+}
+# Checkout to $PRBranch, create new one if not exists.
```

Comment:
What happen to your code to check to see if we were already on that branch? I think that is still needed here to ensure those checkout commands don't fail.

Reply:
Just checked the scenario. 
If we already sit on branch `aBranch`,
run `git show-ref --verify --quiet refs/heads/aBranch` will return 0 for `$LASTEXITCODE`.

Then execute `git checkout $PRBranchName` will return 0 for `$LASTEXITCODE`.

We are good in this case.